gitbutler-app/src/virtual_branches/virtual.rs

```diff
@@ -1799,18 +1800,39 @@ fn set_ownership(
         .filter(|branch| branch.id != target_branch.id)
         .collect::<Vec<_>>();
 
+    let mut branch_claims: Vec<(branch::Branch, Vec<OwnershipClaim>)> = Vec::new();
     for file_ownership in &ownership.claims {
         for branch in &mut virtual_branches {
             let taken = branch.ownership.take(file_ownership);
-            if !taken.is_empty() {
-                branch_writer.write(branch).context(format!(
-                    "failed to write source branch for {}",
-                    file_ownership
-                ))?;
+            branch_claims.push((branch.clone(), taken));
+        }
+    }
+
+    // Check consistency and reject changes if they would result in a file being claimed by multiple branches
```

Comment:
I think s/file/hunk on this line?

But also, just want to make sure we're on the same page. This function won't result in a hunk being claimed by multiple files, instead if two branches claimed the same file this would remove duplicates?

Reply:
yeah, used the wrong word in the comment.

regarding the check - its putting `file-startline-endline` in a hash set, and if that gets repeated errors out, thus asserting that a hunk cant be in two branches at the same time with the update branch operation.

Reply:
Right, makes sense.